integration-tests/tests/src/tests/sync/app.ts

```diff
@@ -0,0 +1,322 @@
+      let eventListenerCalls = 0;
+      expect(this.app).instanceOf(Realm.App);
+
+      const listenerEvent = () => {
+        eventListenerCalls += 1;
+      };
+
+      this.app.addListener(listenerEvent);
+
+      let credentials = Realm.Credentials.anonymous();
+      let user = await this.app.logIn(credentials);
+      expect(eventListenerCalls).equals(1);
+
+      await user.logOut();
+      expect(eventListenerCalls).equals(2);
+      credentials = Realm.Credentials.anonymous();
+
+      user = await this.app.logIn(credentials);
+      expect(eventListenerCalls).equals(3);
+
+      // Remove the listener and verify that the event listener was not fired
+      this.app.removeListener(listenerEvent);
+      await user.logOut();
+      expect(eventListenerCalls).equals(3);
+
+      // Test remove all
+      let eventListener1Calls = 0;
+      let eventListener2Calls = 0;
+      const listenerEvent1 = () => {
+        eventListener1Calls += 1;
+      };
+      const listenerEvent2 = () => {
+        eventListener2Calls += 1;
+      };
+      this.app.addListener(listenerEvent1);
+      this.app.addListener(listenerEvent2);
+
+      user = await this.app.logIn(credentials);
+      expect(eventListener1Calls).equals(1);
+      expect(eventListener2Calls).equals(1);
+
+      this.app.removeAllListeners();
+      await user.logOut();
+      expect(eventListener1Calls).equals(1);
+      expect(eventListener2Calls).equals(1);
+    });
+  });
+
+  describe("with sync", () => {
+    importAppBefore("with-db");
+
+    it("migration while sync is enabled throws", async function (this: Mocha.Context & AppContext & RealmContext) {
+      const user = await this.app.logIn(Realm.Credentials.anonymous());
+      const config = {
+        schema: [TestObjectSchema],
+        sync: { user, partitionValue: '"Lolo"' },
+        deleteRealmIfMigrationNeeded: true,
+      };
+      //@ts-expect-error TYPEBUG: SyncConfiguration interfaces misses a user property.
+      expect(() => new Realm(config)).throws(
+        "Cannot set 'deleteRealmIfMigrationNeeded' when sync is enabled ('sync.partitionValue' is set).",
+      );
+      await user.logOut();
+    });
+    it("MongoDB Realm sync works", async function (this: Mocha.Context & AppContext & RealmContext) {
+      const dogNames = ["King", "Rex"]; // must be sorted
+      let nCalls = 0;
+
+      const credentials = Realm.Credentials.anonymous();
+      const user = await this.app.logIn(credentials);
+      const partition = genPartition();
+      const realmConfig = {
+        schema: [PersonForSyncSchema, DogForSyncSchema],
+        shouldCompact: () => {
+          nCalls++;
+          return true;
+        },
+        sync: {
+          user: user,
+          partitionValue: partition,
+          _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+        },
+      };
+      //@ts-expect-error TYPEBUG: SyncConfiguration interfaces misses a user property.
```

Comment:
I don't think that's the case? https://github.com/realm/realm-js/blob/bindgen/packages/realm/src/app-services/SyncConfiguration.ts#L107